tests/testsuite/lints.rs

```diff
@@ -749,3 +749,163 @@ pub const Ĕ: i32 = 2;
         )
         .run();
 }
+
+#[cargo_test(nightly, reason = "edition2024 is not stable")]
```

Comment:
If possible, I lean toward UI test for specific lints.

(I know this is more like testing the system)

Reply:
I switched to UI tests, let me know if you like the structure